### PR TITLE
Fix react and react-phaser example scripts and add instructions

### PR DESCRIPTION
**Pull request by satyambnsal**
Closes #

#105 
#103 
## Introduced changes
- Fix DojoProvider usage in React examples
- Fix `generate component` script
- Fix `build-phaser` script
- Add React example instructions
- Fix react package build issue

-

## Checklist

-   [ ] Linked relevant issue
-   [x] Updated relevant documentation
-   [ ] Added relevant tests
-   [ ] Add a dedicated CI job for new examples
-   [x] Performed self-review of the code

### CODE REVIEW

File: examples/react/react-app/src/dojo/generated/setup.ts [hunk 1]
```diff
@@ -29,7 +29,11 @@ export async function setup({ ...config }: Config) {
     await getSyncEntities(toriiClient, contractComponents as any);
 
     const client = await setupWorld(
-        new DojoProvider(config.manifest, config.rpcUrl)
+        new DojoProvider(
+            config.worldAddress,
```
Comment (ponderingdemocritus) [inline examples/react/react-app/src/dojo/generated/setup.ts:33]: this shouldn't exist - only the manifest is passed now, have you re built everything?

Comment (satyambnsal) [reply examples/react/react-app/src/dojo/generated/setup.ts:33]: @ponderingdemocritus Yes, I've rebuilt everything from scratch. Also tried once more after seeing your comment and I see that DojoProvider expects world address as first argument.

I used the same syntax in dojo jam app as well. 
https://github.com/satyambnsal/castle-hexapolis/blob/main/client/src/dojo/setupNetwork.ts#L27

Comment (satyambnsal) [reply examples/react/react-app/src/dojo/generated/setup.ts:33]: I see 
when I see the DojoProvider constructor, it expects only manifest and url
```
constructor(manifest?: any, url: string = LOCAL_KATANA) {
        super(manifest.world.address);
        this.provider = new RpcProvider({
            nodeUrl: url,
        });

        this.contract = new Contract(
            manifest.world.abi,
            this.getWorldAddress(),
            this.provider
        );
        this.manifest = manifest;
    }
    ```
    Not sure what's wrong. exploring it bit more.

Comment (satyambnsal) [reply examples/react/react-app/src/dojo/generated/setup.ts]: Not sure what was the issue but after deleting node modules from react app as well as from top-level and rebuilding everything resolves the issue!.

**Pull request merged**